test/integration/targets/azure_rm_networkinterface/tasks/main.yml

```diff
@@ -319,6 +322,139 @@
       - output.state.ip_configurations[0].public_ip_address.name == "tn{{ rpfx }}"
       - output.state.enable_accelerated_networking
 
+- name: Create application security group(check mode)
+  azure_rm_applicationsecuritygroup:
+      resource_group: "{{ resource_group }}"
+      name: "{{ applicationsecuritygroup_name1 }}"
+      tags:
+          testing: testing
+  check_mode: yes
+  register: output
+
+- name: Assert check mode creation
+  assert:
+    that:
+      - output.changed
+
+- name: Create Application security group
+  azure_rm_applicationsecuritygroup:
+    resource_group: "{{ resource_group }}"
+    name: "{{ applicationsecuritygroup_name1 }}"
+    tags:
+        testing: testing
+  register: output
+
+- name: Assert application security group creation
+  assert:
+    that:
+      - output.changed
+      - output.id != ''
+
+- name: Create application security group (idempotent)
+  azure_rm_applicationsecuritygroup:
+      resource_group: "{{ resource_group }}"
+      name: "{{ applicationsecuritygroup_name1 }}"
+      tags:
+          testing: testing
+  register: output
+
+- name: Assert idempotent
+  assert:
+    that:
+      - not output.changed
+
+- name: Update application security group
+  azure_rm_applicationsecuritygroup:
+      resource_group: "{{ resource_group }}"
+      name: "{{ applicationsecuritygroup_name1 }}"
+      tags:
+          testing: testing
+          foo: bar
+  register: output
+
+- name: Assert update
+  assert:
+    that:
+      - output.changed
+
+- name: Create Application security group in secondary resource group
+  azure_rm_applicationsecuritygroup:
+    resource_group: "{{ resource_group_secondary }}"
+    name: "{{ applicationsecuritygroup_name2 }}"
+  register: asg
+
+- name: Create Nic with application security groups
```

Comment:
just wondering if we could reuse one of the nics that was already created?
but that's not that significant, as nic is not very heavy.